Source/Core/Common/MemoryUtil.h

```diff
@@ -14,10 +14,13 @@ void* AllocateMemoryPages(size_t size);
 void FreeMemoryPages(void* ptr, size_t size);
 void* AllocateAlignedMemory(size_t size, size_t alignment);
 void FreeAlignedMemory(void* ptr);
-void ReadProtectMemory(void* ptr, size_t size);
-void WriteProtectMemory(void* ptr, size_t size, bool executable = false);
-void UnWriteProtectMemory(void* ptr, size_t size, bool allowExecute = false);
+bool MemProtect(void* ptr, size_t size, bool read, bool write, bool exec);
 std::string MemUsage();
 size_t MemPhysical();
 
+// Deprecated
+void ReadProtectMemory(void* ptr, size_t size);
```

Comment:
Just an idea, but C++ has the `[[deprecated]]` attribute that would be used like:

```cpp
[[deprecated("newer_func() should be used instead")]]
void some_func();
```